editor/src/components/filebrowser/confirm-branch-disconnect.tsx

```diff
@@ -0,0 +1,66 @@
+/** @jsxRuntime classic */
+/** @jsx jsx */
+import { jsx } from '@emotion/react'
+import React from 'react'
+import { disconnectGithubProjectActions } from '../../core/shared/github'
+import { Dialog, FormButton } from '../../uuiui'
+import { EditorDispatch } from '../editor/action-types'
+import * as EditorActions from '../editor/actions/action-creators'
+
+interface ConfirmDisconnectBranchProps {
+  dispatch: EditorDispatch
+  branchName: string
+}
+
+export const ConfirmDisconnectBranchDialog: React.FunctionComponent<
+  React.PropsWithChildren<ConfirmDisconnectBranchProps>
+> = (props) => {
+  const hide = React.useCallback(() => {
+    props.dispatch([EditorActions.hideModal()], 'everyone')
+  }, [props])
+  return (
+    <Dialog
+      title='Disconnect project from branch'
+      content={<DialogBody {...props} />}
+      defaultButton={<AcceptButton {...props} />}
+      secondaryButton={<CancelButton {...props} />}
+      closeCallback={hide}
+    />
+  )
+}
+
+const DialogBody: React.FunctionComponent<React.PropsWithChildren<ConfirmDisconnectBranchProps>> = (
+  props,
+) => (
+  <React.Fragment>
+    <p>
+      Are you sure you want to disconnect this project from the Github branch{' '}
+      <strong>“{props.branchName}”</strong>?
```

Comment:
suggested addition: "Your project will not be affected by this, but won't be able to save or load from Github until you reconnect to a branch."

Reply:
:+1: https://github.com/concrete-utopia/utopia/pull/2794/commits/b19e5790bc713da79ce33e81aca19b6669ea86ae